tfexec/internal/e2etest/util_test.go

```diff
@@ -41,6 +41,32 @@ func runTest(t *testing.T, fixtureName string, cb func(t *testing.T, tfVersion *
 		versions = strings.Split(override, ",")
 	}
 
+	// If the env var TFEXEC_E2ETEST_TERRAFORM_PATH is set to the path of a
+	// valid Terraform executable, only tests appropriate to that
+	// executable's version will be run.
+	if localBinPath := os.Getenv("TFEXEC_E2ETEST_TERRAFORM_PATH"); localBinPath != "" {
+		td, err := ioutil.TempDir("", "tf")
+		if err != nil {
+			t.Fatalf("error creating temporary test directory: %s", err)
+		}
+		t.Cleanup(func() {
+			os.RemoveAll(td)
+		})
```

Comment:
Is there a reason we have to _create_ a directory for the execution of `terraform version`? Wouldn't it suffice to use  [`os.TempDir()`](https://pkg.go.dev/os#TempDir)?

Reply:
Perhaps this is a little paranoid. The failure case for using `os.TempDir()` is that it contains some Terraform configuration files that somehow in a future version of Terraform affect the result of `terraform version`. I agree this is pretty unlikely. I do, however, like our convention that during tests, every time a new `Terraform{}` is initialised it is in a totally clean directory.

Reply:
Good point - I _think_ it _already_ behaves differently depending on path, because it shows versions of installed providers. So - good call to execute in a clean directory.

Maybe just leave a comment to explain _why_ we're doing that for future ourselves?